scripts/globals/abilities/feather_step.lua

```diff
@@ -0,0 +1,180 @@
+                    daze = 2
+                    effect = 1
+                end
+            end
+        else
+            if (target:hasStatusEffect(tpz.effect.BEWILDERED_DAZE_1)) then
+                local duration = target:getStatusEffect(tpz.effect.BEWILDERED_DAZE_1):getDuration()
+                target:delStatusEffectSilent(tpz.effect.BEWILDERED_DAZE_1)
+                target:addStatusEffect(tpz.effect.BEWILDERED_DAZE_2, 1, 0, duration+30)
+                effect = 2
+
+            elseif (target:hasStatusEffect(tpz.effect.BEWILDERED_DAZE_2)) then
+                local duration = target:getStatusEffect(tpz.effect.BEWILDERED_DAZE_2):getDuration()
+                target:delStatusEffectSilent(tpz.effect.BEWILDERED_DAZE_2)
+                target:addStatusEffect(tpz.effect.BEWILDERED_DAZE_3, 1, 0, duration+30)
+                effect = 3
+
+            elseif (target:hasStatusEffect(tpz.effect.BEWILDERED_DAZE_3)) then
+                local duration = target:getStatusEffect(tpz.effect.BEWILDERED_DAZE_3):getDuration()
+                target:delStatusEffectSilent(tpz.effect.BEWILDERED_DAZE_3)
+                target:addStatusEffect(tpz.effect.BEWILDERED_DAZE_4, 1, 0, duration+30)
+                effect = 4
+
+            elseif (target:hasStatusEffect(tpz.effect.BEWILDERED_DAZE_4)) then
+                local duration = target:getStatusEffect(tpz.effect.BEWILDERED_DAZE_4):getDuration()
+                target:delStatusEffectSilent(tpz.effect.BEWILDERED_DAZE_4)
+                target:addStatusEffect(tpz.effect.BEWILDERED_DAZE_5, 1, 0, duration+30)
+                effect = 5
+
+            elseif (target:hasStatusEffect(tpz.effect.BEWILDERED_DAZE_5)) then
+                local duration = target:getStatusEffect(tpz.effect.BEWILDERED_DAZE_5):getDuration()
+                target:delStatusEffectSilent(tpz.effect.BEWILDERED_DAZE_5)
+                target:addStatusEffect(tpz.effect.BEWILDERED_DAZE_5, 1, 0, duration+30)
+                effect = 5
+            else
+                target:addStatusEffect(tpz.effect.BEWILDERED_DAZE_1, 1, 0, 60)
+                effect = 1
+            end
+        end
+
+        if (player:hasStatusEffect(tpz.effect.FINISHING_MOVE_1)) then
+            player:delStatusEffectSilent(tpz.effect.FINISHING_MOVE_1)
+            player:addStatusEffect(tpz.effect.FINISHING_MOVE_1+daze, 1, 0, 7200)
+
+        elseif (player:hasStatusEffect(tpz.effect.FINISHING_MOVE_2)) then
+            player:delStatusEffectSilent(tpz.effect.FINISHING_MOVE_2)
+            player:addStatusEffect(tpz.effect.FINISHING_MOVE_2+daze, 1, 0, 7200)
+
+        elseif (player:hasStatusEffect(tpz.effect.FINISHING_MOVE_3)) then
+            player:delStatusEffectSilent(tpz.effect.FINISHING_MOVE_3)
+            if (daze > 2) then
+                daze = 2
+            end
+            player:addStatusEffect(tpz.effect.FINISHING_MOVE_3+daze, 1, 0, 7200)
+
+        elseif (player:hasStatusEffect(tpz.effect.FINISHING_MOVE_4)) then
```

Comment:
(I've never played DNC, so I don't know the specifics)

There is a lot of repetition with all of this DNC stuff, is there anything logical we can take advantage of?
Loopifying things, breaking out early wherever possible, (nuclear option) a bunch of status_type or grouping flags so we can hand off this repetition to core etc.

Reply:
So I looked thru older stuff and all seems basically the same. I wish I knew a way to get it to look neater but everything I’ve tried other than the way it is hasn’t worked or didn’t take away/add the right amount of steps or daze/effect